theano/tensor/elemwise.py

```diff
@@ -589,36 +586,13 @@ def make_node(self, *inputs):
                                                     out_broadcastables)]
         return Apply(self, inputs, outputs)
 
-    def __eq__(self, other):
-        if type(self) == type(other):
+    def __str__(self):
+        if self.inplace_pattern:
             items = list(self.inplace_pattern.items())
-            other_items = list(other.inplace_pattern.items())
             items.sort()
-            other_items.sort()
-            rval = ((self.scalar_op == other.scalar_op) and
-                    (items == other_items))
-            return rval
-        return False
-
-    def _rehash(self):
-        inplace_pattern_hash = hash_from_dict(self.inplace_pattern)
-        h = hash('Elemwise') ^ hash(self.scalar_op) ^ inplace_pattern_hash
-        assert h == getattr(self, '_hashval', h)
-        self._hashval = h
-
-    def __hash__(self):
-        return self._hashval
-
-    def __str__(self):
-        if self.name is None:
-            if self.inplace_pattern:
-                items = list(self.inplace_pattern.items())
-                items.sort()
-                return "Elemwise{%s}%s" % (self.scalar_op, str(items))
-            else:
-                return "Elemwise{%s}" % (self.scalar_op)
+            return "Elemwise{%s}%s" % (self.scalar_op, str(items))
         else:
-            return self.name
+            return "Elemwise{%s}" % (self.scalar_op)
```

Comment:
This is a good way. But to lower the change in what we return I would do.
name = self.scalar_op.replace('scalar_', '')

This way elemwise like xlogx won't have its name change.